janus_server/src/aggregator.rs

```diff
@@ -404,7 +433,7 @@ impl TaskAggregator {
                     .clone()
                     .try_into()
                     .map_err(|_| Error::TaskParameters(task::Error::AggregatorAuthKeySize))?;
-                VdafOps::Prio3Aes128Count(vdaf, verify_key)
+                VdafOps::Prio3Aes128Count(Arc::new(vdaf), verify_key)
```

Comment:
Is the `Arc` really necessary here? A VDAF implementation is basically just a bunch of function pointers with virtually no data, so just `.clone()`ing them (without an `Arc`) should be cheap enough. Plus [trait `Vdaf` already requires `Clone`](https://docs.rs/prio/latest/prio/vdaf/trait.Vdaf.html).

Reply:
The justification is that since this VDAF will eventually be passed to code that is generic over the specific VDAF type, we don't (in general) know what a VDAF is. While that's true today, we don't know that some future VDAF doesn't contain a nontrivial amount of state. (precomputed data, perhaps?) It may not ever matter, but it's minor defensive programming IMO.

(Given the above argument, I'd suggest that `Vdaf` shouldn't require `Clone` since we shouldn't be cloning them since we can't guarantee they'll be small; but I don't care enough to try to make that change.)